docs/maintenance/RELEASES.md

```diff
@@ -0,0 +1,85 @@
+1. Add any breaking changes intended for the release to that milestone.
+1. Create two new branches off `main` in the project repository (not a personal fork):
+   - `v${major}`
+   - `v${major}-canary-auto-release`
+1. Send a PR from `v${major}-canary-auto-release` to `main` modifying the [`ci.yml` workflow](https://github.com/typescript-eslint/typescript-eslint/blob/main/.github/workflows/ci.yml) [example: [chore: add auto-canary release for v6](https://github.com/typescript-eslint/typescript-eslint/pull/5883)]:
+   - Under `pushes:` at the beginning of the file, add an `- v${major}` list item.
+   - Add a `publish_canary_version_v${major}` step the same as `publish_canary_version` except:
+     - Add the condition: `if: github.ref == 'refs/heads/v${major}'`.
+     - Its publish command should be `npx lerna publish premajor --loglevel=verbose --canary --exact --force-publish --yes --dist-tag rc-v${major}`.
+
+#### 2. Merging Breaking Changes
+
+1. Send a PR from `v${major}` to `main` [example: [v6.0.0](https://github.com/typescript-eslint/typescript-eslint/pull/5886)].
+1. Change all [breaking change PRs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+is%3Aopen+label%3A%22breaking+change%22) to target the `v${major}` branch.
+   - To signify these changes as breaking, the first line of the PR description must read as `BREAKING CHANGE:`, and second line should briefly summarize the changes.
+   - It is important to note that when merged the commit message must also include `BREAKING CHANGE:` as the first line in order for lerna to recognize it as a breaking change in the release notes. If you miss this it just means more manual work when writing the release documentation.
+1. Wait until all required PRs have been merged
+1. Let the release wait for **at least 1 week** to allow time for early adopters to help test it and discuss the changes.
+   - Promote it on the [`@tseslint`](https://twitter.com/tseslint) twitter to get some additional attention.
+1. Once discussions have settled, rebase merge the PR on top of `main`.
+
+:::note
+_Non_-breaking changes can be merged to `main` or the major branch.
+They don't need any special treatment.
+:::
+
+#### 3. Releasing the Version
+
+1. Discuss with the maintainers to be ready for an [out-of-band](#out-of-band) release. Doing this manually helps ensure someone is on-hand to action any issues that might arise from the major release.
+1. Prepare the release notes. Lerna will automatically generate the release notes on GitHub, however this will be disorganized and unhelpful for users. We need to reorganize the release notes so that breaking changes are placed at the top to make them most visible. If any migrations are required, we must list the steps to make it easy for users.
```

Comment:
Add some example links for easy access and inspiration

```suggestion
1. Prepare the release notes. Lerna will automatically generate the release notes on GitHub, however this will be disorganized and unhelpful for users. We need to reorganize the release notes so that breaking changes are placed at the top to make them most visible. If any migrations are required, we must list the steps to make it easy for users.
  - Example release notes: [`v5.0.0`](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.0.0), [`v4.0.0`](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0), [`v3.0.0`](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v3.0.0)
```